File: tuan2/PhanB/bai42.jl
function main()
    tokens = split(join(readlines(), " "))
    if isempty(tokens) return end
    n, k = parse(Int, tokens[1]), parse(Int, tokens[2])
    if k > n || k <= 0 return end
    a = [parse(Int, tokens[i]) for i in 3:n+2]
    s = sum(a[1:k])
    max_sum = s
    best_start = 1
    for i in (k + 1):n
        s += a[i] - a[i - k]
        if s > max_sum
            max_sum = s
            best_start = i - k + 1
        end
    end
    println("tong $max_sum, bat dau tai vi tri $best_start")
end
main()

File: tuan2/PhanB/bai42.py
import sys

def main():
    data = sys.stdin.read().split()
    if not data: return
    n, k = int(data[0]), int(data[1])
    if k > n or k <= 0: return
    a = [int(x) for x in data[2:n+2]]
    s = sum(a[:k])
    max_sum = s
    best_start = 0
    for i in range(k, n):
        s += a[i] - a[i - k]
        if s > max_sum:
            max_sum = s
            best_start = i - k + 1
    print(f"tong {max_sum}, bat dau tai vi tri {best_start + 1}")

if __name__ == '__main__':
    main()